spec/controllers/api/v3/medical_histories_controller_spec.rb

```diff
@@ -74,4 +74,121 @@ def create_record_list(n, options = {})
+        expect(history.diabetes_boolean).to eq(expected_values[:diabetes_boolean])
+        expect(history.diagnosed_with_hypertension_boolean).to eq(expected_values[:diagnosed_with_hypertension_boolean])
+        expect(history.prior_heart_attack).to eq(expected_values[:prior_heart_attack])
+        expect(history.prior_stroke).to eq(expected_values[:prior_stroke])
+        expect(history.chronic_kidney_disease).to eq(expected_values[:chronic_kidney_disease])
+        expect(history.receiving_treatment_for_hypertension).to eq(expected_values[:receiving_treatment_for_hypertension])
+        expect(history.diabetes).to eq(expected_values[:diabetes])
+        expect(history.diagnosed_with_hypertension).to eq(expected_values[:diagnosed_with_hypertension])
+        expect(history.hypertension).to eq(expected_values[:hypertension])
+        expect(history.receiving_treatment_for_diabetes).to eq(expected_values[:receiving_treatment_for_diabetes])
+      end
+
+      it "sets values to nil when not provided" do
+        params[:medical_histories][0].delete(:receiving_treatment_for_diabetes)
+
+        post :sync_from_user, params: params
+
+        expect(response.status).to eq 200
+        history = patient.medical_history
+        expected_values = params[:medical_histories].first
+
+        expect(history.id).to eq(expected_values[:id])
+        expect(history.patient_id).to eq(expected_values[:patient_id])
+        expect(history.prior_heart_attack_boolean).to eq(expected_values[:prior_heart_attack_boolean])
+        expect(history.prior_stroke_boolean).to eq(expected_values[:prior_stroke_boolean])
+        expect(history.chronic_kidney_disease_boolean).to eq(expected_values[:chronic_kidney_disease_boolean])
+        expect(history.receiving_treatment_for_hypertension_boolean).to eq(expected_values[:receiving_treatment_for_hypertension_boolean])
+        expect(history.diabetes_boolean).to eq(expected_values[:diabetes_boolean])
+        expect(history.diagnosed_with_hypertension_boolean).to eq(expected_values[:diagnosed_with_hypertension_boolean])
+        expect(history.prior_heart_attack).to eq(expected_values[:prior_heart_attack])
+        expect(history.prior_stroke).to eq(expected_values[:prior_stroke])
+        expect(history.chronic_kidney_disease).to eq(expected_values[:chronic_kidney_disease])
+        expect(history.receiving_treatment_for_hypertension).to eq(expected_values[:receiving_treatment_for_hypertension])
+        expect(history.diabetes).to eq(expected_values[:diabetes])
+        expect(history.diagnosed_with_hypertension).to eq(expected_values[:diagnosed_with_hypertension])
+        expect(history.hypertension).to eq(expected_values[:hypertension])
+        expect(history.receiving_treatment_for_diabetes).to eq(nil)
+      end
+
+      it "returns 200 but returns errors and does not create a medical history when provided invalid values" do
```

Comment:
👍 I agree this is surprising behavior, and probably something we want to change at some point, but good to have it documented in an explicit spec like this.